### PR TITLE
validate mutation query results 

### PR DESCRIPTION
**Pull request by ani-sha**
This PR contains (in reference #16) : 
- Support for GraphQL Mutations 
- Fetching corrected results 
- Support for multiple queries to issue results after mutations
- Basic test cases

### CODE REVIEW

File: engine/src/main/java/io/graphqlcrud/SQLDataFetcher.java [hunk 1]
```diff
@@ -35,34 +36,80 @@
...
+
+    private ResultSet executeSQL(DataFetchingEnvironment environment, SQLContext ctx) throws Exception {
         String sql = buildSQL(environment);
         ctx.setSQL(sql);
         LOGGER.info("SQL Executed:" + sql);
 
         ResultSet rs = null;
         Connection c = ctx.getConnection();
-        Statement stmt = c.createStatement();
+        Statement stmt = c.createStatement(ResultSet.TYPE_SCROLL_INSENSITIVE,ResultSet.CONCUR_READ_ONLY);
```
Comment (rareddy) [inline engine/src/main/java/io/graphqlcrud/SQLDataFetcher.java:95]: why do you need to set this? this is an expensive operation also not all jdbc drivers support this

File: engine/src/test/java/io/graphqlcrud/SQLMutationDataFetcherTest.java [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+                "  \"g0\".\"LASTNAME\" \"LASTNAME\",\n" +
+                "  \"g0\".\"PHONE\" \"PHONE\"\n" +
+                "from PUBLIC.CUSTOMER \"g0\"\n" +
+                "where \"g0\".\"SSN\" = 'CST01040'\n" +
+                "order by \"g0\".\"SSN\"";
+
+        Assertions.assertEquals(create_expected_mutation,create_result.get(1));
+        Assertions.assertEquals(create_expected_sql, create_result.get(0));
+
+        String update_query = "mutation {\n" +
```
Comment (rareddy) [inline engine/src/test/java/io/graphqlcrud/SQLMutationDataFetcherTest.java:102]: split into different test cases create, update, delete

File: engine/src/main/java/io/graphqlcrud/SQLDataFetcher.java (hunk 1 above)
Comment (ani-sha) [reply engine/src/main/java/io/graphqlcrud/SQLDataFetcher.java:95]: Oops realized that now :/ 
I was trying to get the no. of rows returned after query execution to validate delete operation.

**Comment by ani-sha** [comment]
> What is the difference between `primary_key` vs `primary_fields`? I think `primary_fields` was designed for this

 I overlooked as we haven't used `primary_fields` anywhere, will take that into use.

**Comment by ani-sha** [comment]
> In "which" situations you would make `Input` PK value as part of the condition? seems contradictory to me from one side, is good when a filter is not supplied, but only picking out primary fields does not sound right, and one can not select all the fields as a condition.

`input` would be used for a `create`. So should we drop the idea of using PK for conditions? Rather go along with taking `and` of all provided values as condition?

**Comment by ani-sha** [comment]
> I believe `primaryKey` and `primaryFields` same here?

yes both are same.

**Comment by rareddy** [comment]
> > In "which" situations you would make `Input` PK value as part of the condition? seems contradictory to me from one side, is good when a filter is not supplied, but only picking out primary fields does not sound right, and one can not select all the fields as a condition.
> 
> `input` would be used for a `create`. So should we drop the idea of using PK for conditions? Rather go along with taking `and` of all provided values as condition?

There is no condition in the "insert" SQL command, so it should not apply if you are only using for the "create"

**Comment by ani-sha** [comment]
> I think we need to allow the "Address" without PK, because that is possible in a relational database. There will be one to many cases where many side just there because of the other side. Here Address means nothing without a customer. That means the `SSN` is not null field than it has it's own primary key. What we can do is any table that does not have a primary key, we should not have the top level methods like `address, createAddress, updateAddress and deleteAddress` then we can handle them later with nested Customer.

I thought defining a PK was mandatory. 

> > > In "which" situations you would make `Input` PK value as part of the condition? seems contradictory to me from one side, is good when a filter is not supplied, but only picking out primary fields does not sound right, and one can not select all the fields as a condition.
> > 
> > 
> > `input` would be used for a `create`. So should we drop the idea of using PK for conditions? Rather go along with taking `and` of all provided values as condition?
> 
> There is no condition in the "insert" SQL command, so it should not apply if you are only using for the "create"

After an insert, the select query uses the `input` PK value to define the condition.

**Comment by ani-sha** [comment]
> > I thought defining a PK was mandatory.
> 
> It is not mandatory, the implementation can choose to say we only want to support tables with PK, but that would be wrong IMO as we would not able to make them change the DDL for their databases. We just need to change how we are going to support that case.
> 
> > After an insert, the select query uses the input PK value to define the condition.
> 
> By not allowing the top-level queries (only customer, not address) this case will be still true, As Customer will have the PK.

Well in that case i will remove the PK from `address`, and need to do some research about correct the way to approach for multiple queries.

**Review by rareddy — APPROVED** [review]
LGTM

**Pull request merged**